lib/features/email/domain/usecases/move_to_mailbox_interactor.dart

```diff
@@ -26,16 +29,23 @@ class MoveToMailboxInteractor {
       final currentEmailState = listState.last;
 
       final result = await _emailRepository.moveToMailbox(session, accountId, moveRequest);
+
       if (result.isNotEmpty) {
         yield Right(MoveToMailboxSuccess(
-          result.first,
-          moveRequest.currentMailboxes.keys.first,
-          moveRequest.destinationMailboxId,
-          moveRequest.moveAction,
-          moveRequest.emailActionType,
-          destinationPath: moveRequest.destinationPath,
-          currentMailboxState: currentMailboxState,
-          currentEmailState: currentEmailState));
+            result.first,
+            moveRequest.currentMailboxes.keys.first,
+            moveRequest.destinationMailboxId,
+            moveRequest.moveAction,
+            moveRequest.emailActionType,
+            destinationPath: moveRequest.destinationPath,
+            currentMailboxState: currentMailboxState,
+            currentEmailState: currentEmailState));
+        if (moveRequest.emailActionType == EmailActionType.moveToSpam){
+          Set<EmailId> emailIds = Set.from(result);
+          final emails = _emailRepository.getListDetailedEmailById(session, accountId, emailIds);
+          List<Email> emailList = await emails;
+          _emailRepository.markAsRead(session, accountId, emailList, ReadActions.markAsRead);
```

Comment:
@davopo Thank for your PR. But IMO, This is not a good solution, we have to call too many requests to the jmap server. Instead we just need to update the `keywords/$seen` property in the `update` properties of `SetEmailMethod`. You can fix it in the `moveToMailbox` function of `EmailAPI`.

Reply:
Thank you, i will do it.

Reply:
@dab246 what do you think ? Are you proposing any other solutions ?

Reply:
@davopo Please see below comment